www/include/configuration/configKnowledge/display-services.php

```diff
@@ -45,28 +45,21 @@
 require_once $centreon_path . '/bootstrap.php';
 $pearDB = $dependencyInjector['configuration_db'];
 
-if (!isset($limit) || !$limit) {
+if (!isset($limit) || !$limit || $limit < 0) {
     $limit = $centreon->optGen["maxViewConfiguration"];
 }
 
-if (isset($_POST['searchHost'])) {
-    if (!isset($_POST['searchHasNoProcedure']) && isset($_GET['searchHasNoProcedure'])) {
-        unset($_REQUEST['searchHasNoProcedure']);
-    }
-    if (!isset($_POST['searchTemplatesWithNoProcedure']) && isset($_GET['searchTemplatesWithNoProcedure'])) {
-        unset($_REQUEST['searchTemplatesWithNoProcedure']);
-    }
-}
-
 $order = "ASC";
 $orderby = "host_name";
-if (isset($_REQUEST['order'])
-    && $_REQUEST['order']
-    && isset($_REQUEST['orderby'])
-    && $_REQUEST['orderby']
-) {
-    $order = $_REQUEST['order'];
-    $orderby = $_REQUEST['orderby'];
+
+// Use whitelist as we can't bind ORDER BY values
+if (!empty($_POST['order']) && !empty($_POST['orderby'])) {
+    if (in_array($_POST['order'], ["ASC", "DESC"])) {
+        $order = $_POST['order'];
+    }
+    if (in_array($_POST['orderby'], ["host_name", "service_description"])) {
```

Comment:
could a third argument be used ?

Reply:
are you sure that a $_GET['orderBy'] couldn't be used too ?
as previously a $_REQUEST was checked and on line 99 a search.php script is required